### ISSUE TITLE
jj status stalls for 10 seconds after printing output

### ISSUE THREAD

**Issue by edre** [martinvonz/jj#3628]
## Description

In a large-ish [git repository](https://gitlab.alpinelinux.org/alpine/aports) (20k files, 200k linear commits), `jj status` takes ~1 second to snapshot the working copy, then does seemingly nothing at 100% cpu for 10 seconds after printing the output.

Linux perf says 25% of cpu time is coming from miniz_oxide::inflate::core::decompress, with stacktrace:

```
jj_cli::commands::run_command                                                                                                                                            
        <alloc::vec::Vec<T> as alloc::vec::spec_from_iter::SpecFromIter<T,I>>::from_iter                                                                             
        <core::iter::sources::from_fn::FromFn<F> as core::iter::traits::iterator::Iterator>::next                                                                                      
        <jj_lib::default_index::rev_walk::FilterRevWalk<W,P> as jj_lib::default_index::rev_walk::RevWalk<I>>::next                                                                     
        jj_lib::default_index::revset_engine::build_predicate_fn::_$u7b$$u7b$closure$u7d$$u7d$::hef4c6357b53a6f23                                                                      
        jj_lib::store::Store::get_commit                                                                                                                                     
        _$LT$jj_lib..git_backend..GitBackend$u20$as$u20$jj_lib..backend..Backend$GT$::read_commit::_$u7b$$u7b$closure$u7d$$u7d$::hf7175803741d19ab                               
        _$LT$jj_lib..git_backend..GitBackend$u20$as$u20$jj_lib..backend..Backend$GT$::read_commit::_$u7b$$u7b$closure$u7d$$u7d$::_$u7b$$u7b$closure$u7d$$u7d$::h8adf5db9f0d4bea2     
        gix::repository::object::<impl gix::types::Repository>::_find_object_inner_generated_by_gix_macro_momo                                                                  
        gix_odb::cache::impls::<impl gix_pack::find_traits::Find for gix_odb::Cache<S>>::try_find_cached                                                                       
        gix_odb::store_impls::dynamic::find::<impl gix_odb::store_impls::dynamic::Handle<S>>::try_find_cached_inner                                                                     
```

This is the FilterRevWalk used to compute [ancestor_conflicts](https://github.com/martinvonz/jj/blob/20af8c79ef9edcdcdef3bada20d1887529662c71/cli/src/commands/status.rs#L98).

Maybe jj should stop searching through ancestors when we find an immutable ancestor?

## Steps to Reproduce the Problem

```
jj git clone https://gitlab.alpinelinux.org/alpine/aports
cd aports
jj status
```

## Expected Behavior

Instantaneous command completion after printing the "Parent commit" line.

## Actual Behavior

Seconds of uselessly burning CPU.

## Specifications

- Platform: Linux, static musl release build
- Version: 0.17.0

**Comment by thoughtpolice** [comment]
This repository is really not that large, I regularly use `jj` (with watchman) on large repos like `nixpkgs`. However I happened to notice a discrepancy between my installed version of `jj`, which completes `jj st` in 0.1s, versus the 0.17.0 release, which takes 1.8s on my `--release` build when compiled on my machine. My installed version is a customized version of `jj` using f58e69f6 as a base, committed on April 19th. A build based on f58e69f6 consistently completes in 0.1s for me.  This is also true of v0.16.0, so I consider this a regression.

**Comment by martinvonz** [comment]
When did you start using jj with that repo (`jj git clone`, `jj init --git-dir`, etc.)? If you started with jj < [0.11.0](https://github.com/martinvonz/jj/releases/tag/v0.11.0), then this is quite expected. You may want to recreate your repo in that case.

Not checking for conflicts on immutable commits still sounds like a good idea to me.

**Comment by edre** [comment]
This code was added in https://github.com/martinvonz/jj/commit/52e494dcf2653012da6d64984c7e0496b5a0f856, which was merged just before 0.17.0 was cut.

**Comment by martinvonz** [comment]
> This code was added in [52e494d](https://github.com/martinvonz/jj/commit/52e494dcf2653012da6d64984c7e0496b5a0f856), which was merged just before 0.17.0 was cut.

Sounds like we want to release a 0.17.1 with either a fix or a rollback of that commit then. Thanks for letting us know.
